signer/tests/integration/communication.rs

```diff
@@ -1,182 +1,125 @@
-    // of connected peers and wait until we have the expected number.
-    tokio::time::sleep(Duration::from_secs(2)).await;
 
-    // Test that trusted 2 can send a message to trusted 1.
-    let trusted_msg_from_2_to_1 = tokio::time::timeout(Duration::from_secs(1), async {
-        trusted1.receive().await.unwrap();
-    });
-    trusted2
-        .broadcast(Msg::random_with_private_key(&mut rand::thread_rng(), &key2))
-        .await
-        .unwrap();
-    trusted_msg_from_2_to_1
-        .await
-        .expect("Failed to receive message from trusted 2 to trusted 1");
-
-    // Test that trusted 1 can send a message to trusted 2.
-    let trusted_msg_from_1_to_2 = tokio::time::timeout(Duration::from_secs(1), async {
-        trusted2.receive().await.unwrap();
-    });
-    trusted1
-        .broadcast(Msg::random_with_private_key(&mut rand::thread_rng(), &key1))
-        .await
-        .unwrap();
-    trusted_msg_from_1_to_2
-        .await
-        .expect("Failed to receive message from trusted 1 to trusted 2");
-
-    // Test that adversarial can't send a message to trusted 1.
-    let adversarial_msg_to_1 = tokio::time::timeout(Duration::from_secs(1), async {
-        trusted1.receive().await.unwrap();
-    });
-    adversarial
-        .broadcast(Msg::random(&mut rand::thread_rng()))
-        .await
-        .unwrap();
-    assert!(adversarial_msg_to_1.await.is_err());
-
-    // Test that adversarial can't send a message to trusted 2.
-    let adversarial_msg_to_2 = tokio::time::timeout(Duration::from_secs(1), async {
-        trusted2.receive().await.unwrap();
-    });
-    adversarial
-        .broadcast(Msg::random(&mut rand::thread_rng()))
-        .await
-        .unwrap();
-    assert!(adversarial_msg_to_2.await.is_err());
-
-    // Kill the swarms just to be sure.
-    handle1.abort();
-    handle2.abort();
-    handle3.abort();
+    // Wait for the swarms to start.
+    tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
+
+    let swarm1_addr = swarm1.listen_addrs().await.pop().unwrap();
+    let swarm2_addr = swarm2.listen_addrs().await.pop().unwrap();
+
+    dbg!(&swarm1_addr);
+    dbg!(&swarm2_addr);
+
+    swarm1.dial(swarm2_addr).await.unwrap();
+    swarm2.dial(swarm1_addr).await.unwrap();
+
+    tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
+
+    // Run the test with a 30-second timeout for the swarms to exchange messages.
+    if let Err(_) = tokio::time::timeout(
+        tokio::time::Duration::from_secs(30),
+        signer::testing::network::assert_clients_can_exchange_messages(
+            network1, network2, key1, key2,
+        ),
+    )
+    .await
+    {
+        handle1.abort();
+        handle2.abort();
+        panic!(
+            r#"Test timed out, we waited for 30 seconds but this usually takes around 5 seconds.
+        This is generally due to connectivity issues between the two swarms."#
+        );
+    }
```

Comment:
if now this test is stable, can we lower the timeout?

Reply:
Lowered it to 10 [28906f0](https://github.com/stacks-network/sbtc/pull/1242/commits/28906f0e0c7e49266640b4573dedbb6680b324b0)